content/en/blog/_posts/2022-11-11-retroactive-default-storage-class.md

```diff
@@ -0,0 +1,131 @@
+
+| Without default class        | PVC storageClassName == `""` | PVC storageClassName == `nil` |
+|------------------------------|------------------------------|-------------------------------|
+| PV storageClassName == `""`  | binds                        | binds                         |
+| PV without storageClassName  | binds                        | binds                         |
+
+
+| With default class          | PVC storageClassName == `""` | PVC storageClassName == `nil` |
+|-----------------------------|------------------------------|-------------------------------|
+| PV storageClassName == `""` | binds                        | class updates                 |
+| PV without storageClassName | binds                        | class updates                 |
+
+## How to use it
+
+You don't have to perform any configuration steps - the feature will work "out of the box" once it's GA.
+
+If you want to test the feature in Alpha you need to enable FeatureGates in `kube-controller-manager` and `kube-apiserver`:
+
+```bash
+--feature-gates="...,RetroactiveDefaultStorageClass=true"
+```
+
+### Test drive
+
+If you would like to see the feature in action and verify it works fine in your cluster here's what you can try:
+
+1. We'll use a very simple PersistentVolumeClaim.
+```yaml
+apiVersion: v1
+kind: PersistentVolumeClaim
+metadata:
+  name: csi-pvc
+spec:
+  accessModes:
+  - ReadWriteOnce
+  resources:
+    requests:
+      storage: 1Gi
+```
+
+2. Create the PersistentVolumeClaim when there is no default StorageClass. The PVC won't provision or bind (unless there is an existing, suitable PV already present) and will remain in `Pending` state.
+```bash
+$ kc get pvc
+NAME      STATUS    VOLUME   CAPACITY   ACCESS MODES   STORAGECLASS   AGE
+csi-pvc   Pending   
+```
+
+3. Configure one StorageClass as default.
+
+```bash
+$ kc patch sc/csi-hostpath-sc -p '{"metadata":{"annotations":{"storageclass.kubernetes.io/is-default-class":"true"}}}'
+storageclass.storage.k8s.io/csi-hostpath-sc patched
+```
+
+4. Verify that PersistentVolumeClaims is now provisioned correctly and was updated retroactively with new default StorageClass.
+
+```bash
+$ kc get pvc
+NAME      STATUS   VOLUME                                     CAPACITY   ACCESS MODES   STORAGECLASS      AGE
+csi-pvc   Bound    pvc-06a964ca-f997-4780-8627-b5c3bf5a87d8   1Gi        RWO            csi-hostpath-sc   87m
+```
+
+### New metrics
+
+To help indicate that the feature is working as expected we also introduced new `retroactive_storageclass_total` metric to show how many times PV controller attempted to update PersistentVolumeClaim and `retroactive_storageclass_errors_total` to show how many of those attempts failed.
+
+## Getting involved
+
+If you would like to share feedback with us, you can do so on our [public Slack channel](https://app.slack.com/client/T09NY5SBT/C09QZFCE5).
```

Comment:
Explain who "us" means. So far, you've made it sound like "us" would mean Red Hat as you haven't mentioned SIG Storage clearly enough for a reader to make the link.

Reply:
This needs to be addressed, please.